server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java

```diff
@@ -212,6 +222,40 @@ public IssueQuery createFromRequest(SearchWsRequest request) {
     }
   }
 
+  private Date buildCreatedAfterFromRequest(DbSession dbSession, SearchWsRequest request, Set<String> componentUuids, boolean effectiveOnComponentOnly) {
+    Date createdAfter = parseAsDateTime(request.getCreatedAfter());
+    String createdInLast = request.getCreatedInLast();
+
+    if (request.getSinceLeakPeriod() == null || !request.getSinceLeakPeriod()) {
+      return buildCreatedAfterFromDates(createdAfter, createdInLast);
+    }
+
+    checkRequest(createdAfter == null, "'%s' and '%s' cannot be set simultaneously", CREATED_AFTER, SINCE_LEAK_PERIOD);
+    Set<String> allComponentUuids = new HashSet<>(componentUuids);
+    if (!effectiveOnComponentOnly) {
+      if (request.getProjectKeys() != null) {
+        allComponentUuids.addAll(componentUuids(dbSession, request.getProjectKeys()));
+      }
+      if (request.getProjectUuids() != null) {
+        allComponentUuids.addAll(request.getProjectUuids());
+      }
+      if (request.getModuleUuids() != null) {
+        allComponentUuids.addAll(request.getModuleUuids());
+      }
+      if (request.getFileUuids() != null) {
+        allComponentUuids.addAll(request.getFileUuids());
+      }
+    }
+
+    checkArgument(allComponentUuids.size() == 1, "One and only one component must be provided when searching since leak period");
+    String uuid = allComponentUuids.iterator().next();
+    // TODO use ComponentFinder instead
+    ComponentDto component = checkFoundWithOptional(componentService.getByUuid(uuid), "Component with id '%s' not found", uuid);
+    SnapshotDto snapshot = dbClient.snapshotDao().selectLastSnapshotByComponentId(dbSession, component.getId());
+    Date createdAfterFromSnapshot = snapshot == null ? null : longToDate(snapshot.getPeriodDate(1));
+    return buildCreatedAfterFromDates(createdAfterFromSnapshot, createdInLast);
+  }
```

Comment:
This class is really a mess, I hope we'll be able to sanitize it when rails will be removed as it's hard to read/maintain